packages/eslint-plugin/docs/rules/no-wrapper-object-types.mdx

```diff
@@ -9,15 +9,20 @@ import TabItem from '@theme/TabItem';
 >
 > See **https://typescript-eslint.io/rules/no-wrapper-object-types** for documentation.
 
-The JavaScript language has a set of language types, but some of them correspond to two TypeScript types, which look similar: `boolean`/`Boolean`, `number`/`Number`, `string`/`String`, `bigint`/`BigInt`, `symbol`/`Symbol`, `object`/`Object`.
-The difference is that the lowercase variants are compiler intrinsics and specify the actual _runtime types_ (that is, the return value when you use the `typeof` operator), while the uppercase variants are _structural types_ defined in the library that can be satisfied by any user-defined object with the right properties, not just the real primitives.
-JavaScript also has a "wrapper" class object for each of those primitives, such as [`Boolean`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) and [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number).
-These wrapper objects are assignable to the uppercase types, but not to the lowercase types.
-
-Using the primitives like `0` instead of object wrappers like `new Number(0)` is generally considered a JavaScript best practice.
-JavaScript programs typically work with the real number primitives, rather than objects that "look like" numbers.
-Primitives are simpler to conceptualize and work with `==` and `===` equality checks -- which their object equivalents do notDeepEqual.
-As a result, using the lowercase type names like `number` instead of the uppercase names like `Number` helps make your code behave more reliably.
+The JavaScript language has a set of built-in ("intrinsic") types which look similar: `boolean`/`Boolean`, `number`/`Number`, `string`/`String`, `bigint`/`BigInt`, `symbol`/`Symbol`, and `object`/`Object`.
```

Comment:
I think this loses the nuance that we tried hard to keep in the original docs. JavaScript only has eight types: `number`, `object`, `string`, `symbol`, `boolean`, `bigint`, `undefined`, `null`. `Boolean`, `Number`, etc. are subtypes of `object`. It's only in TypeScript-land where they are sort of parallel and can be compared to each other.

Reply:
Neither here nor there, but 

> `Boolean`, `Number`, etc. are subtypes of `object`.

TIL that 
```ts
let o: object = false; // <-- TS error
// BUT
let b: Boolean = false;
o = b; // no error lol
``` 

The wrapper types are so bad lol

Reply:
Yep—everything that doesn't claim to be a primitive is implicitly `object`. `object` does not actually mean "non-primitive"; it just means "is not definitely primitive".